mozilla/blurts-server
# Add cronjob for the monthly free email

# References:

Jira: MNTOR-3606
Figma:



# Description

There's quite a lot in this PR, as I found the need to make more changes as I tested the end-to-end flow - apologies for it getting unwieldy. What I did:

- Remove the need for an explicit `React` import (but since this is broken on `main`, I've also submitted that separately: https://github.com/mozilla/blurts-server/pull/5153)
- Removed the `MonthlyActivityEmail` flag
- Removed the unsubscribe link button from the email admin tool, and added the ability to send the activity email (which contains that link)
- Fixed the unsubscription API call
- Fixed the unsubscription API for cases where the subscription preference isn't set
- Updated Storybook to show MJML errors
- Fix MJML errors in the free monthly email template
- Grouped the monthly emails into `monthlyActivityFree` and `monthlyActivityPlus` folders, i.e. at the same level in the hierarchy
- And of course, added the new cron job

I'll go through the PR to call out some things as well :)

# How to test

You might want to add this to your `.env.local` to only do two at a time, to make it easier to verify in the DB for example.

This should run the cron job directly: `npm run dev:cron:monthly-activity-free`

This should be more realistic for production: `npm run build-cronjobs; npm run cron:monthly-activity-free`

That should send the activity email to free users whose account was created later than 1 January 2022, and whose language was set to `en-US` when signing up (or was unset).

# Checklist (Definition of Done)

- [ ] I've added a unit test to test for potential regressions of this bug. - Lots of side effects
